grype/match/ignore.go

```diff
@@ -21,6 +23,7 @@ type IgnoreRule struct {
 	Namespace     string            `yaml:"namespace" json:"namespace" mapstructure:"namespace"`
 	FixState      string            `yaml:"fix-state" json:"fix-state" mapstructure:"fix-state"`
 	Package       IgnoreRulePackage `yaml:"package" json:"package" mapstructure:"package"`
+	Pom           IgnoreRulePom     `yaml:"pom" json:"pom" mapstructure:"pom"`
```

Comment:
I wonder if this should be nested under a Java ignore rule container? E.g. the configuration would be:
```yaml
  # ...or just by a single pom field:
  - java:
      pom:
        scope: test
```

Reply:
@jneate the team talked about this PR a bit and had a few comments:

The first thing to note is 

The simplest change to get this to the finish line: since this is for a package rather than something else we believe this should be nested under the `IgnoreRulePackage`, probably for now as `java` (since the metadata is `pkg.JavaMetadata` rather than Maven POM specific), so the configuration would look something like:
```yaml
- package:
    java:
      scope: 'test'
      artifact-id: 'log4j'
      group-id: 'org.apache.logging.log4j'
```
... I think this would basically just involve moving the
```
Pom           IgnoreRulePom     `yaml:"pom" json:"pom" mapstructure:"pom"`
```
under the `IgnoreRulePackage`, as something more like:
```
Java           IgnoreRuleJava     `yaml:"java" json:"java" mapstructure:"java"`
```

--

But... I'd also mention there was another thought that we might like to make this more generic, where we could potentially specify any metadata information, something like:
```yaml
- package:
    metadata:
      - field : .PomScope
        value: test
      - field :  .PomArtifactID
        value: log4j
      - field :  .PomGroupID
        value: 'org.apache.logging.log4j'
```

Do you think either of these options would be something you'd like to do? If not, we could probably help at least get the first change done to get this to the finish line.